### DISCUSSION TITLE
Migration like for seeds

### DISCUSSION THREAD

**Discussion by aneeskhan47** [laravel/framework#38586] [Ideas]
Hi,

In this approach, each seeders is a class, like a migration: it defines both "up" and "down" methods, so you can run and rollback your seeds. Like migrations, the order of the files is determined by a timestamp at the creation time.

My use case for this is to populate my apps in production, in order to automatize this process, without having to deal with running each seed class individually, and manually.

What do you think?

**Comment by mikield** [comment]
Hello,

Do you want to have track of seeders as migrations too?

IHMO, you have 2 options to resolve such problem:
1) Create a "Seeder" class with 2 static methods: `up` and `down`, and use them into your migration file.
2) You can just code your "data migration" right after the schema changes.

Using any of suggested options will record a migration into your database, where scheme and data was changed.
